File: src/main/java/gr/jimmys/jimmysfoodzilla/common/Result.java
package gr.jimmys.jimmysfoodzilla.common;

public record Result(boolean isSuccess, String error, Object ResultValue) {
    public static Result success() {
        return new Result(true, null, null);
    }

    public static Result success(Object result) {
        return new Result(true, null, result);
    }

    public static Result failure(String error) {
        return new Result(false, error, null);
    }
}

File: ReactMeals_WebApi/Common/Result.cs
namespace ReactMeals_WebApi.Common
{
    public record Result(bool IsSuccess, string Error)
    {
        public static Result Success() => new(true, null);
        public static Result Failure(string error) => new(false, error);
    }

    public record Result<T>(bool IsSuccess, string Error, T Value)
    {
        public static Result<T> Success(T value) => new(true, null, value);
        public static Result<T> Failure(string error) => new(false, error, default);
    }
}
